### PR TITLE
Differentiability of hyperelastic periodic simulation (homogenization)

### PR DESCRIPTION
**Pull request by Huangzizhou**

### CODE REVIEW

File: src/polyfem/mesh/Mesh.cpp [hunk 1]
```diff
@@ -250,14 +250,14 @@ namespace polyfem::mesh
...
 		std::vector<int> tmp(cells.data(), cells.data() + cells.size());
-		std::sort(tmp.begin(), tmp.end());
-		tmp.erase(std::unique(tmp.begin(), tmp.end()), tmp.end());
-
-		mesh->in_ordered_vertices_ = Eigen::Map<Eigen::VectorXi, Eigen::Unaligned>(tmp.data(), tmp.size());
-		// assert(mesh->in_ordered_vertices_[0] == 0);
-		// assert(mesh->in_ordered_vertices_[1] == 1);
-		// assert(mesh->in_ordered_vertices_[2] == 2);
-		// assert(mesh->in_ordered_vertices_[mesh->in_ordered_vertices_.size() - 1] == vertices.rows() - 1);
+ 		std::sort(tmp.begin(), tmp.end());
```
Comment (teseoch) [inline src/polyfem/mesh/Mesh.cpp:253]: this is my change, right?

File: src/polyfem/solver/forms/adjoint_forms/VariableToSimulation.cpp [hunk 2]
```diff
@@ -24,11 +27,38 @@ namespace polyfem::solver
...
+		else if (type == "periodic-shape")
+			return std::make_unique<PeriodicShapeVariableToSimulation>(states, parametrization);
 
 		log_and_throw_adjoint_error("Invalid type of VariableToSimulation!");
 		return std::unique_ptr<VariableToSimulation>();
 	}
 
+	void VariableToSimulation::set_output_indexing(const json &args)
+	{
+		const std::string composite_map_type = args["composite_map_type"];
```
Comment (teseoch) [inline src/polyfem/solver/forms/adjoint_forms/VariableToSimulation.cpp:39]: wy do you need this?

**Review by teseoch — APPROVED** [review]

File: src/polyfem/mesh/Mesh.cpp (hunk 1 above)
Comment (Huangzizhou) [reply src/polyfem/mesh/Mesh.cpp:253]: This is only spacing...Should I revert it?

File: src/polyfem/solver/forms/adjoint_forms/VariableToSimulation.cpp (hunk 2 above)
Comment (Huangzizhou) [reply src/polyfem/solver/forms/adjoint_forms/VariableToSimulation.cpp:39]: This is moving around the inverse optimization code, the branching was ugly so I moved into each derived class

**Pull request merged**